webapp/src/components/rhs/user_badge_row.tsx

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+
+import Client4 from 'mattermost-redux/client/client4';
+
+import {UserBadge} from '../../types/badges';
+import BadgeImage from '../utils/badge_image';
+import {markdown} from 'utils/markdown';
+
+import './user_badge_row.scss';
+
+type Props = {
+    badge: UserBadge;
+    isCurrentUser: boolean;
+    onClick: (badge: UserBadge) => void;
+}
+
+const UserBadgeRow: React.FC<Props> = ({badge, onClick, isCurrentUser}: Props) => {
+    const time = new Date(badge.time);
+    let reason = null;
+    if (badge.reason) {
+        reason = (<div className='badge-user-reason'>{'Why? ' + badge.reason}</div>);
+    }
+    let setStatus = null;
+    const c = new Client4();
+    if (isCurrentUser && badge.image_type === 'emoji') {
+        setStatus = (<div className='user-badge-set-status'><a onClick={() => c.updateCustomStatus({emoji: badge.image, text: badge.name})}>{'Set status to this badge'}</a></div>);
```

Comment:
Can we make it multiline, looks a bit forced here to put everything in one line.